workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/utilities/NonFungibleTokenBuilder.kt

```diff
@@ -0,0 +1,69 @@
+package com.r3.corda.lib.tokens.workflows.utilities
+
+import com.r3.corda.lib.tokens.contracts.states.NonFungibleToken
+import com.r3.corda.lib.tokens.contracts.types.IssuedTokenType
+import com.r3.corda.lib.tokens.contracts.types.TokenType
+import com.r3.corda.lib.tokens.contracts.utilities.heldBy
+import com.r3.corda.lib.tokens.contracts.utilities.issuedBy
+import com.r3.corda.lib.tokens.workflows.TokenBuilderException
+import net.corda.core.contracts.Amount
+import net.corda.core.identity.Party
+
+/**
+ * A utility class designed for Java developers to more easily access Kotlin DSL
+ * functions to build non-fungible tokens and their component classes.
+ *
+ * This function vaguely follows a builder pattern, design choices were made
+ * to emulate Kotlin syntax as closely as possible for an easily transferable
+ * developer experience in Java.
+ */
+class NonFungibleTokenBuilder {
```

Comment:
ny reason why we can't use the same builder for both fungible and non-fungible tokens? Both of the builders seem to have the same fields...

Reply:
This was a usability compromise. Requested this so that it's more intuitive for novice users to quickly discern what elements they need to add to build fungible vs non-fungible.

Reply:
ok makes sense